internal/daemon/worker/proxy/proxydialer.go

```diff
@@ -19,19 +22,70 @@ var GetEndpointDialer = directDialer
 
 // directDialer returns a ProxyDialer which tcp dials directly to the provided
 // endpoint.
-func directDialer(ctx context.Context, endpoint string, _ string, _ proto.Message, _ interface{}) (*ProxyDialer, error) {
+func directDialer(ctx context.Context, endpoint string, _ string, _ proto.Message, _ interface{}, opt ...Option) (*ProxyDialer, error) {
 	const op = "proxy.directDialer"
 	if len(endpoint) == 0 {
 		return nil, errors.New(ctx, errors.InvalidParameter, op, "endpoint is empty")
 	}
-	d, err := NewProxyDialer(ctx, func(opt ...Option) (net.Conn, error) {
+	opts := GetOpts(opt...)
+	dnsServer := opts.WithDnsServer
+	parsedDnsServer, err := url.Parse(dnsServer)
+	if err != nil {
+		return nil, errors.Wrap(ctx, err, op)
+	}
+
+	d, err := NewProxyDialer(ctx, func(dialerOpt ...Option) (net.Conn, error) {
```

Comment:
Does this function benefit from being a closure? Wonder if it'd be slightly easier to read if it was a named function

Reply:
Maybe, but it's not a change from this PR and I want to keep it scoped to the new stuff.